src/grading-settings/GradingSettings.jsx

```diff
@@ -0,0 +1,181 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { Container, Layout, Button } from '@edx/paragon';
+import { CheckCircle, Warning } from '@edx/paragon/icons';
+
+import AlertMessage from '../generic/alert-message';
+import { RequestStatus } from '../data/constants';
+import InternetConnectionAlert from '../generic/internet-connection-alert';
+import SubHeader from '../generic/sub-header/SubHeader';
+import { getGradingSettings, getSavingStatus, getLoadingStatus } from './data/selectors';
+import { fetchGradingSettings, sendGradingSetting } from './data/thunks';
+import GradingScale from './grading-scale/GradingScale';
+import GradingSidebar from './grading-sidebar';
+import messages from './messages';
+import { getGradingValues, getSortedGrades } from './grading-scale/utils';
+
+const GradingSettings = ({ intl, courseId }) => {
+  const gradingSettingsData = useSelector(getGradingSettings);
```

Comment:
Let's consider your proposal in the next stage of working on the page when it will be clear what kind of code we have as a result. This is a good idea, thanks 👍